server/controllers/review.js

```diff
@@ -0,0 +1,44 @@
+const Review = require('../models/Review');
+const asyncHandler = require("express-async-handler");
+
+// @route GET /review
+// @desc Get all reviews of a user
+// @access Public
+exports.userReviews = asyncHandler(async (req, res, next) => {
+  const { uid } = req.params;
+
+  const reviews = await Review.find({ userId: uid });
+
+  if (!reviews) {
+    res.status(404);
+    throw new Error('No reviews found for this user.');
+  }
+
+  res.status(200).json({ reviews: reviews });
```

Comment:
great json return!

Reply:
About the error return, I am trying to keep consistent with the patterns in other controllers, but I do think an empty array or a message with success code is better since we haven't actually encountered an error, I will update it.

Reply:
Updated this one